xapian-core/cluster/kmeans.cc

```diff
@@ -130,13 +130,13 @@ KMeans::initialize_kmeanspp(ClusterSet &cset) {
 }
 
 void
-KMeans::initialize_points(MSetDocumentSource source, TermListGroup &tlg) {
+KMeans::initialize_points(const MSet &source, TermListGroup &tlg) {
     LOGCALL_VOID(API, "KMeans::initialize_points()", source | tlg);
-    while (!source.at_end()) {
-	Document temp = source.next_document();
+    MSetIterator it = source.begin();
+    for(; it != source.end(); it++) {
+	Document temp = it.get_document();
```

Comment:
Yes I had removed them in most other parts. Didn't see this. Thanks for pointing out!